### ISSUE TITLE
Better plot coloring / market price visual

### ISSUE THREAD

**Issue by pmaji** [pmaji/crypto-whale-watching-app#44]
- [x] show logic for margins specification and adding background color
- [ ] need to choose a better background / plot color (just picked one arbitrarily to show proof of concept)
- [x] need to add horizontal line to show market price 
- [ ] would love to be to selectively color the background such that above the market price is a very faint color of red and below market price is a very faint color of green (this comes after the horizontal line implementation)

**Comment by theimo1221** [comment]
For horizontal line
`'shapes': [ {'type': 'line',
            'x0': 1,
            'y0': 0,
            'x1': 1,
            'y1': 2,
            'line': { 'color': 'rgb(255, 255, 255)', 'width': 3 },`

For background color I´ll try some JS.

**Comment by pmaji** [comment]
Good deal. I just pushed the horizontal line to master. If you can do the background color with JS that'd be great @theimo1221  :)

**Comment by theimo1221** [comment]
I don´t like the hardcided suze, I mean look at this:
![image](https://user-images.githubusercontent.com/20605452/36364138-8e7136d8-1542-11e8-944a-a78682cbe555.png)

We just waste space we have...

**Comment by theimo1221** [comment]
What is wrong with this:
Vertical screen:
![image](https://user-images.githubusercontent.com/20605452/36364247-429a1378-1543-11e8-8d01-fc7c8f71310a.png)
Horicontal Screen:
![image](https://user-images.githubusercontent.com/20605452/36364276-66b41dda-1543-11e8-9dea-01d44ee2c580.png)

**Comment by theimo1221** [comment]
I can add you some JS to set it to fix size, but  having it responsive on window resizes and stuffs needs to stay active....

**Comment by pmaji** [comment]
@theimo1221 good call. I hadn't thought about the JS. Just changed master back to use autosizing.

**Comment by theimo1221** [comment]
But keeping the margin in seems good for me.  Version #48 looks quite good.

**Comment by pmaji** [comment]
Good deal. Margin spacing is quite nice. All that remains for this issue is any iteration we want on the background color.

**Comment by theimo1221** [comment]
#59 opens possibilities to improve this one

**Comment by pmaji** [comment]
Integrated #59 @theimo1221 
-JS option now good to go. I want to keep that to just the aesthetic alterations. Top priority is some visual cleaning up and hopefully a dark-mode version. What I'm linking below looks viable, but let me know your thoughts.

https://stackoverflow.com/questions/28357405/dark-mode-switch

**Comment by theimo1221** [comment]
I don´t wan´t to toggle it static over css.
I´d like to do it [this](https://bgrins.github.io/spectrum/) way.
I´ll add a spoiler hide/ showing a div in wich you can set Color for multiple obejcts.
Afterwards chosen color get´s stored in localstorage,  to be able to maintain individual user settings

**Comment by Cygnusfear** [comment]
Hi, I like the concept behind this app but I have some UI/UX questions:

- What is the reasoning behind the current axes?
- I am curious as to whether it is more usable if this would be overlayed on the actual depth chart, I have a feeling it would be easier to process visually
- Are you able to identify 'unique' whales? I figure it would make sense to color code them, so we can see if they have both sell/buy walls (this would also be very cool on a depth chart)
- I could see the benefit in 'ghosts', showing walls that have been pulled at half opacity for a few updates

I am not a Python programmer (I am fine in javascript), if you are interested in the above features I would like to help.

**Comment by theimo1221** [comment]
Axis: 
Price obvious volume for highlighting the influence 

Depth chart 
Adressed in #20 accidentally closed 

Unique whale:
Not by user but sometimes we can guess it by sizes. But we don't want to present assumption just clear facts 

Ghosts:
Currently not implemented, but obviously you can just press freeze open a 2nd tab and have 2nd instance 

As your points mainly address plotly js help would be needed in other components

**Comment by theimo1221** [comment]
@Cygnusfear feel free to review #70

**Comment by pmaji** [comment]
Taken care of. Only thing that remains before this can be closed is the background color / darkmode question. I care less about the selective coloring as mentioned in my initial bullet points.

**Comment by theimo1221** [comment]
Yeah, I will do it. But I'm quite busy currently at both my jobs...
Weekend might be a good time to implement some more

**Comment by pmaji** [comment]
@theimo1221 haha no rush man! I feel you on both fronts. Like I laid out on our call recently, I think we're in a really good place. Just steady improvements from here, but our baseline is solid.

**Comment by mifunetoshiro** [comment]
The last bullet point might be done with two rectangle shapes in Plotly: https://github.com/plotly/plotly.js/issues/922#issuecomment-246358503

**Comment by theimo1221** [comment]
@mifunetoshiro not what we are looking for, due to the fact we want it changeable by the user.
So changing it in Plotly would change it for all people visiting the site.

So we have to do it with JS on clientside

**Comment by mifunetoshiro** [comment]
The way I discovered https://github.com/pmaji/crypto-whale-watching-app/issues/89 is I was trying to plot a horizontal line of the market price and the change from 24h ago like this:

![image](https://i.imgur.com/ZDCXdy8.png)

I noticed that the line wouldn't draw on the BTC-GBP and LTC-BTC plots for some reason (and still doesn't, even with @theimo1221's fix https://github.com/pmaji/crypto-whale-watching-app/pull/90)

Here is my hacky code: https://github.com/mifunetoshiro/crypto-whale-watching-app/commit/a8fcbc43ae41b843be60371db5edfae7e7177a0f (throws '288' key errors in console for some reason)

It also says on the GDAX API reference that "Historical rates should not be polled frequently." Ah well.

There's also a bug in GDAX's get_product_historic_rates API for BCH-BTC and BCH-EUR pairs when using "granularity" 300. It is supposed to return UNIX timestamps in 300 timeslices, e.g. 1520003700, 1520004000, 1520004300, etc., but it doesn't, so the BCH-BTC and BCH-EUR plots aren't even drawn with the code above (index error). I've already sent an email to Coinbase about that.

Anyway @pmaji maybe this is something that could be implemented by someone with better programming skills than me :D

**Comment by theimo1221** [comment]
@mifunetoshiro I like the idea but in my opinion it´s not for the "Main" Purpose of this application.

We shouldn´t forget, we are a "Whale Watching App". Our purpose is to spot and highlight whale behaviour.

As @pmaji already declared: "Whale watching should be used aside to Trading charts like cryptowatch."

So in fact we give that information those platforms are lagging regarding whales.
And showing a line for 24/1 h History is in my opinion a nice feature but useless for hunting whales.

Besides, I´d like to keep further computation low, cause even those 12 Pairs in a scalable code produce quite some CPU load.

**Comment by mifunetoshiro** [comment]
> And showing a line for 24/1 h History is in my opinion a nice feature but useless for hunting whales.

Agree 100%, it's just that cryptowatch doesn't give me this info and I wanted it for myself, and was wondering how the bubbles moved from the price x hours ago.

**Comment by theimo1221** [comment]
@mifunetoshiro cryptowatch is giving the information, you can draw a line from yesterdays price.

Or look at https://coinmarketcap.com it gives you all you need:
![image](https://user-images.githubusercontent.com/20605452/36937074-21ff37ea-1f0e-11e8-90e9-32fac16b657e.png)

**Comment by pmaji** [comment]
@mifunetoshiro I understand the desire for the additional information, but @theimo1221 summarized my view well--we aren't trying to replicate info that people can get from other sources--we want to remain whale-focused. That said, it would also be computationally and visually difficult to add a sustainable 24 hour MP line. I'll give just one reason of many here. 

When we calculate our price margins (i.e. how high and low the y-axis goes above/below market price, we do this based on a percentage calculation. So say our threshold is that the y axis spans $40 above and below market price, but yesterday the price was $100 lower than it is today. This would lead to a hidden line, or a forceably distorted axis. 

tl;dr : it's a good idea, but a bit too computationally complicated, and not aligned with the main purpose of the app. Hope that makes sense.

**Comment by theimo1221** [comment]
> This would lead to a hidden line
In fact plotly would try to draw all elements resulting in a different axis

**Comment by mifunetoshiro** [comment]
> you can draw a line from yesterdays price.
I haven't figured out how to draw a line that would show me the price and percentage difference on cryptowatch...

**Comment by theimo1221** [comment]
They have a button for drawing objects with all stuff like fibonnaci percentage lines curves and all other stuff

**Comment by mifunetoshiro** [comment]
I know that, but I didn't figure out how to get what I wanted with any of those.

**Comment by theimo1221** [comment]
![image](https://user-images.githubusercontent.com/20605452/36958455-41cc8cba-203c-11e8-8f25-c9d18fc2c941.png)

**Comment by mifunetoshiro** [comment]
Yes, but it doesn't tell me the price and percentage change xD

**Comment by theimo1221** [comment]
Cryptonaut gives you price and percentage change for 24 hours week month and year

**Comment by theimo1221** [comment]
Even for your complete portfolio

**Comment by mifunetoshiro** [comment]
> @mifunetoshiro not what we are looking for, due to the fact we want it changeable by the user.
> So changing it in Plotly would change it for all people visiting the site.

Ok, regarding the last bullet point: I think the current plot color (red/green) is the most appropriate. Every exchange lists buy/sell orders as green and red, and except for the colorblind option, I don't really see the need to have this user-configurable - I doubt a lot of people would change it. As for the current dark grey background, I think it'd be better if the was lighter @pmaji

**Comment by pmaji** [comment]
Good ideas @mifunetoshiro ; I've gone with a lighter color for the plot background color. Still some contrast, but I think easier on the eyes.

**Comment by pmaji** [comment]
These have both now been taken care of
